doc/pmon/shutdown_and_startup_fabric_design.md

```diff
@@ -0,0 +1,67 @@
+  * [2.3 Modify the ModuleUpdater in chassisd.](#23-modify-the-moduleupdater-in-chassisd)
+- [3 Test Considerations](#3-test-considerations)
+- [4 References](#4-references)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+###### Revision
+| Rev |     Date    |       Author                                                                       | Change Description                |
+|:---:|:-----------:|:----------------------------------------------------------------------------------:|-----------------------------------|
+| 1.0 | 05/03/2024  | Marty Lok                                                                   | Initial public version            |
+
+# About this Manual
+This document describes the requirements for reset System Fabric Module (SFM)  on a Chassis and the planned design/code changes for supporting this enhancement.
+
+# Scope
+This scope of this specification is using the existing module shutdown/startup commands and the modification supports resetting of the SFM Module
+
+# Background
+In order to avoid the crash of the Swss/Syncd processes of a SFM module, when a SFM module in a chassis is required to be reseated or hot swapped, a proper shutdown and startup procedure needs to be followed.  
+
+
+# 1 Requirements Overview
+## 1.1 Functional Requirements
+This section describes the requirement for using existing CLI command shutdown/startup a SFM module on a Chassis system.  
+1. Using the existing CLI command "sudo config chassis module shutdown/startup <module_name>" to shutdown/startup a SFM module
+2. Module remains down state if system is booting up with a configuration file which contains a module is set to down state
+
+# 2 Design Details
+The following changes are implementation and modification.
+## 2.1 Using the existing CLI command "sudo config chassis module shutdown/startup <module_name>" by modifying/enhancing the chassis_module.py 
+1. Define and create a new method fabric_module_set_admin_status() with the following actions
+  * Derive a list of ASIC number (asic_list) which is assoicated with this module_name from the CHASSIS_FABRIC_ASIC_TABLE in the CHASSIS_STATE_DB
+  * For shutdown case:
+    - Loop this asic_list and call the "systemctl stop" to stop the related swss@ and syncd@ services.
+    - Delete the related CHASSIS_FABRIC_ASIC_TABLE entries in the CHASSIS_STATE_DB
+    - Loop this asic_list and call the "systemctl start" to start the related swss@ and syncd@ service.  The association of service ASIC number with a SFM module is platform specified. If we don't restart of the service here, we are not able to derive the asic_list which is assoicted with this Module when user issues CLI command to start up this SFM module since CHASSIS_FABRIC_ASIC_TABLE entry has been deleted. 
+    
+  * For startup case:
+    - Loop the asic_list and call the "systemctl start" to start the related swss@ and syncd@ service
```

Comment:
same comment as above_

Reply:
This is on Supervisor card.  teamd and bgp are not enabled.   Start the swss service, it will start the syncd service and all other related dependent services if they are applicable.